server/src/main/java/org/elasticsearch/health/HealthPeriodicLogger.java

```diff
@@ -0,0 +1,257 @@
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.health;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.lucene.util.SetOnce;
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.cluster.ClusterChangedEvent;
+import org.elasticsearch.cluster.ClusterStateListener;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.component.Lifecycle;
+import org.elasticsearch.common.logging.ESLogMessage;
+import org.elasticsearch.common.scheduler.SchedulerEngine;
+import org.elasticsearch.common.scheduler.TimeValueSchedule;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.gateway.GatewayService;
+import org.elasticsearch.health.node.selection.HealthNode;
+
+import java.io.Closeable;
+import java.time.Clock;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * This class periodically logs the results of the Health API to the standard Elasticsearch server log file.
+ */
+public class HealthPeriodicLogger implements ClusterStateListener, Closeable, SchedulerEngine.Listener {
+    public static final String HEALTH_FIELD_PREFIX = "elasticsearch.health";
+
+    /**
+     * Creates a new HealthPeriodicLoggerResult.
+     * This creates a record that's used as the output of this feature.
+     *
+     * @param indicatorResults the results of the Health API call that will be used as the output.
+     */
+    public record HealthPeriodicLoggerResult(List<HealthIndicatorResult> indicatorResults) {
+
+        /**
+         * Create a Map of the results, which is then turned into JSON for logging
+         */
+        public Map<String, Object> toMap() {
+            final Map<String, Object> result = new HashMap<>();
+
+            if (this.indicatorResults != null && this.indicatorResults.size() > 0) {
+                // overall status
+                final HealthStatus status = HealthStatus.merge(this.indicatorResults.stream().map(HealthIndicatorResult::status));
+                result.put("elasticsearch.health.overall.status", status.xContentValue());
+
+                // top-level status for each indicator
+                this.indicatorResults.forEach((indicatorResult) -> {
+                    result.put(
+                        String.format(Locale.ROOT, "%s.%s.status", HEALTH_FIELD_PREFIX, indicatorResult.name()),
+                        indicatorResult.status().xContentValue()
+                    );
+                });
+            }
+
+            return result;
+        }
+    }
+
+    public static final String HEALTH_PERIODIC_LOGGER_POLL_INTERVAL = "health_periodic_logger.poll_interval";
```

Comment:
The cluster settings usually have a certain structure so similar settings will be grouped together, for example, the health related settings look like this:
```
 "health" : {
      "node" : {
        "enabled" : "true"
      },
      "master_history" : {
        "has_master_lookup_timeframe" : "30s",
        "identity_changes_threshold" : "4",
        "no_master_transitions_threshold" : "4"
      },
      "reporting" : {
        "local" : {
          "monitor" : {
            "interval" : "30s"
          }
        }
      }
    },
```

I would recommend that we structure this setting in a similar way, for example, `health.periodic_logger.poll_interval`. What do you think?